src/jesse_lib.erl

```diff
@@ -139,31 +139,27 @@ normalize_and_sort_check_object(Value, DraftVersion) ->
 %%   in the list.
 %% - Basic JSON types. In that case, we just return the value.
 %% @private
--spec normalize_and_sort_non_object(Value :: any(), DraftVersion :: 3 | 4) -> any().
-normalize_and_sort_non_object({Key, Val}, DraftVersion) ->
-  {Key, normalize_and_sort_check_object(Val, DraftVersion)};
-normalize_and_sort_non_object(Value, DraftVersion) when is_list(Value) ->
-  [normalize_and_sort_check_object(X, DraftVersion) || X <- Value];
-normalize_and_sort_non_object(Value, DraftVersion) ->
-  case DraftVersion of
-    3 -> Value;
-    4 -> case Value of
-           Number when is_number(Number) -> float(Number);
-           _ -> Value
-         end
-  end.
+-spec normalize_and_sort_non_object(Value :: any()) -> any().
+normalize_and_sort_non_object({Key, Val}) ->
+  {Key, normalize_and_sort_check_object(Val)};
+normalize_and_sort_non_object(Value) when is_list(Value) ->
+  [normalize_and_sort_check_object(X) || X <- Value];
+normalize_and_sort_non_object(Value) when is_number(Value) ->
+  float(Value);
```

Comment:
I know JSON specification is not 100% clear about the numbers and some languages/implementations can handle big integers differently:

```javascript
> JSON.parse('{"bignum": 123456789012345678901234567890}')
Object { bignum: 1.2345678901234568e+29 }
```

```python
>>> import json
>>> json.loads('{"bignum": 123456789012345678901234567890}')
{'bignum': 123456789012345678901234567890}
```

But I think in Erlang world most of JSON libraries are fine with big integers in JSON:

```erlang
> jsx:decode(<<"{\"k\":123456789012345678901234567890}">>).
#{<<"k">> => 123456789012345678901234567890}
> jiffy:decode(<<"{\"k\":123456789012345678901234567890}">>, [return_maps]).
#{<<"k">> => 123456789012345678901234567890}
```

but conversion from big integer to float would loose some precision:
```erlang
> A = 123456789012345678901234567890, B = 123456789012345678901234567891.
123456789012345678901234567891
> A =:= B.
false
> float(A) =:= float(B).
true

```